### PR TITLE
[POC] Resource extension has been rewritten

### PR DESCRIPTION
**Pull request by aRn0D**
@pjedrzejewski It is just a POC, it does not work. What do you about the concept?

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Cool idea! :+1: Resource will be a useful monster soon thanks to all our ideas combined! :dancer:

**Comment by aRn0D** [comment]
@pjedrzejewski I thought about creating Compiler pass which will find service by tags. Instead of extending the `SyliusResourcesExtension`to overwrite the method `registerExtension`we can register them a service with a specific tags. What do you think about it ?

**Comment by aRn0D** [comment]
@pjedrzejewski what do you think about the last comment ?

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Extension/RegisterControllerExtension.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\DatabaseDriverFactory;
+use Sylius\Component\Resource\Exception\Driver\InvalidDriverException;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class RegisterControllerExtension 
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isSupported($configurationure)
```
Comment (amenophis) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Extension/RegisterControllerExtension.php]: $configurationure => why ?

Comment (aRn0D) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Extension/RegisterControllerExtension.php]: because I failed...

Comment (Shine-neko) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Extension/RegisterControllerExtension.php]: @amenophis you do not understand the style :D

Comment (amenophis) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Extension/RegisterControllerExtension.php]: :+1:

**Comment by pjedrzejewski** [comment]
I like this Arnaud! :+1: Let's have a look at it after my #2255 is merged. /cc @aRn0D

**Comment by aRn0D** [comment]
it is just a POC! I think it is not the right way, something like #2622 is better. I will work on it when you will merge #2255.

**Comment by pjedrzejewski** [comment]
Closing this for now. We will think about it if there is time for further ResourceBundle refactorings.

**Pull request closed without merging**